cmake/mason.cmake

```diff
@@ -90,17 +90,15 @@ function(mason_use _PACKAGE)
                 set(_URL "${MASON_REPOSITORY}/${_SLUG}.tar.gz")
                 message("[Mason] Downloading package ${_URL}...")
 
-                set(_FAILED)
-                set(_ERROR)
-                # Note: some CMake versions are compiled without SSL support
+                set(_STATUS)
                 get_filename_component(_CACHE_DIR "${_CACHE_PATH}" DIRECTORY)
                 file(MAKE_DIRECTORY "${_CACHE_DIR}")
-                execute_process(
-                    COMMAND curl --retry 3 -s -f -S -L "${_URL}" -o "${_CACHE_PATH}.tmp"
```

Comment:
Using an external `curl` command might seem weird, but we're using this technique rather than the built-in version because there are some CMake versions out there (e.g. the default version provided by Ubuntu Precise) that _don't_ support SSL because they were compiled without OpenSSL. We've also found that cURL's retry support is useful in environments with spotty internet connectivity and it'd be cool to replicate this behavior with CMake's `file(DOWNLOAD ...)` function.

OTOH, we could require users to provide a reasonably modern version of CMake with SSL support, given that compiling on Ubuntu Precise will likely cause more pain than just the lack of SSL in CMake.

Reply:
Ubuntu Precise hit EOL last year, so yeah, probably more pain that just the lack of SSL in CMake :).  I might be able to detect the presence of curl using find_program() and only fall back to file() if it isn't available. Would that work?

Reply:
Yup, that sounds like a good strategy too!